be/src/util/timezone_utils.cpp

```diff
@@ -22,15 +22,59 @@
 
 #include "util/timezone_utils.h"
 
+#include <charconv>
+#include <string_view>
+#include <utility>
+#include <vector>
+
+#include "glog/logging.h"
+#include "util/phmap/phmap.h"
+
 namespace starrocks {
 
 RE2 TimezoneUtils::time_zone_offset_format_reg(R"(^[+-]{1}\d{2}\:\d{2}$)");
 
 const std::string TimezoneUtils::default_time_zone = "+08:00";
 
-bool TimezoneUtils::find_cctz_time_zone(const std::string& timezone, cctz::time_zone& ctz) {
+static phmap::flat_hash_map<std::string_view, cctz::time_zone> _s_cached_timezone;
+static phmap::flat_hash_map<std::pair<std::string_view, std::string_view>, int64_t> _s_cached_offsets;
+
+void TimezoneUtils::init_time_zones() {
+    // timezone cache list
+    // We cannot add a time zone to the cache that contains both daylight saving time and winter time
+    static std::vector<std::string> timezones = {"UTC", "Asia/Shanghai", "Asia/Hong_Kong"};
```

Comment:
Better add `America/Los_Angeles`?

Reply:
https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
UTC offset and UTC DST offset for `America/Los_Angeles` was not the same. so we couldn't add it to timezone offset list.
but we could add it to `other_timezones`